alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/ConfigField.java

```diff
@@ -166,6 +170,22 @@ public void setSubGroup(final String subGroup) {
         this.subGroup = subGroup;
     }
 
+    public Set<String> getRequiredRelatedFields() {
+        return requiredRelatedFields;
+    }
+
+    public void setRequiredRelatedFields(final Set<String> requiredRelatedFields) {
+        this.requiredRelatedFields = requiredRelatedFields;
+    }
+
+    public void requireField(final String configFieldKey) {
```

Comment:
you could have this follow the builder pattern where it returns the this pointer and rather than void it returns ConfigField.  This way the requiring of the fields can be done as an inline operation rather than saving a local variable and calling the local variable requireField.